velox/functions/prestosql/tests/ArrayAnyMatchTest.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Format.h>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+
+class ArrayAnyMatchTest : public functions::test::FunctionBaseTest {
+ protected:
+  // Evaluate an expression.
+  void testExpr(
+      const std::vector<std::optional<bool>>& expected,
+      const std::string& lambdaExpr,
+      const VectorPtr& input) {
+    auto expression = folly::sformat("any_match(c0, x -> ({}))", lambdaExpr);
+    auto result = evaluate(expression, makeRowVector({input}));
+    assertEqualVectors(makeNullableFlatVector<bool>(expected), result);
+  }
+
+  template <typename T>
+  void testExpr(
+      const std::vector<std::optional<bool>>& expected,
+      const std::string& lambdaExpr,
+      const std::vector<std::vector<std::optional<T>>>& input) {
+    auto expression = folly::sformat("any_match(c0, x -> ({}))", lambdaExpr);
```

Comment:
perhaps, `testAnyMatch(expected, lambdaExpr, makeNullableArrayVector(input))` to reduce code duplication